src/Microsoft.ML.PipelineInference/AutoInference.cs

```diff
@@ -158,7 +158,8 @@ private bool GetDataVariableName(IExceptionContext ectx, string nameOfData, JTok
                     return false;
 
                 string dataVar = firstNodeInputs.Value<String>(nameOfData);
-                ectx.Check(VariableBinding.IsValidVariableName(ectx, dataVar), $"Invalid variable name {dataVar}.");
+                if (!VariableBinding.IsValidVariableName(ectx, dataVar))
+                    throw ectx.ExceptParam($"Invalid variable name {dataVar}.");
```

Comment:
>($"Invalid variable name {dataVar}."); [](start = 42, length = 38)

Note that this method takes *two* methods. The first is the name of the variable, the second is the actual error message; we are putting the exception message into the variable name, which is surely a mistake.

I suppose the error is on the parameter `firstNodeInputs`, so maybe `nameof(firstNodeInputs)`.

Reply:
Thanks, fixing.